src/pages/resultsView/coExpression/CoExpressionVizUtils.ts

```diff
@@ -54,16 +55,23 @@ export function computePlotData(
     };
     for (const datum of mutationData) {
         if (datum.proteinChange) {
-            const targetData = (datum.entrezGeneId === xEntrezGeneId ? xMutations : yMutations);
+            const targetData = (datum.entrezGeneId === xGeneticEntityId ? xMutations : yMutations);
             targetData[datum.uniqueSampleKey] = targetData[datum.uniqueSampleKey] || [];
             targetData[datum.uniqueSampleKey].push(datum);
             addSampleInfo(datum);
         }
     }
+
     for (const datum of molecularData) {
-        const targetData = (datum.entrezGeneId === xEntrezGeneId ? xData : yData);
-        targetData[datum.uniqueSampleKey] = datum;
-        addSampleInfo(datum);
+        if ((datum as NumericGeneMolecularData).entrezGeneId !== undefined) {
```

Comment:
it would be a lot more efficient (and readable IMO) to check once outside the loop whether this is a geneset or gene situation, rather than checking once per datum in the loop.